src/main/java/games/strategy/engine/random/PlainRandomSource.java

```diff
@@ -38,10 +30,15 @@ private static long getSeed() {
   }
 
   @Override
-  public synchronized int getRandom(final int max, final String annotation) throws IllegalArgumentException {
-    if (random == null) {
-      random = new MersenneTwister(getSeed());
+  public int getRandom(final int max, final String annotation) {
+    checkArgument(max > 0, "max must be > 0");
+
+    synchronized (classLock) {
+      if (random == null) {
+        random = new MersenneTwister();
+      }
```

Comment:
Is there a reason MersenneTwister is not being initialized on static initialization of the class?
the random field seems to be only set here once...

Reply:
Reading your post on this object:
Given that this was introduced so early, probably no one knows why it was implemented like this...
There are 7 occurences in the code where an instance of PlainRandomSource is created.
Interestingly enough, this object is also being used as random source for the odds calculator, so in theory using the odds calculator in a local game changes the outcome of future dice rolls with the current code.
This wouldn't happen if the complete class would be non-static, a good enhancement IMO

Reply:
Yeah, the lazy initialization (which, again, was there from the beginning) was probably another premature optimization.  I'll convert `random` to an instance field and verify it doesn't cause any issues.